### ISSUE TITLE
(inactive) massconverter

### ISSUE THREAD

**Issue by jaspershen** [Bioconductor/Contributions#2482]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/tidymass/massconverter

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [ ] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by vjcitn** [comment]
Should depend on R >= 4.2.  No vignettes folder.  Seems to be a pkgdown site source.  Please read developer guidelines.

**Comment by jaspershen** [comment]
Hi. Just added vignettes. For the R version, I am using R 4.1, so I am wondering if I can keep 4.1 in my depends? Thanks!

**Comment by jaspershen** [comment]
Hi. Just changed R > 4.2 and ran well. Thanks.

**Comment by vjcitn** [comment]
```
-- re-building ‘convert_data.Rmd’ using rmarkdown
Warning in normalizePath(input_path) :
  path[1]="demo_data/raw_data": No such file or directory
Warning in normalizePath(output_path) :
  path[1]="demo_data/mzxml": No such file or directory
```
Is something missing?

**Comment by jaspershen** [comment]
Thanks @vjcitn . Just fixed the bug in one function and now the warnings have gone! Thank you!

**Comment by lgatto** [comment]
Hi @jaspershen - I am reviewing your packages and wanted to make sure I don't miss any. I'm aware of `massconverter` and `masstools` (issue #2481); anything else?

**Comment by jaspershen** [comment]
Thanks so much, Laurent! That’s all. No other packages. Thank you!

---------------------------
Xiaotao Shen, PhD 
Postdoctoral Research Fellow
Prof. Michael Snyder’s lab<http://med.stanford.edu/snyderlab.html>
Stanford University School of Medicine
Co-chair | SCPA<https://scpa.netlify.app/>
Biomedical Innovations Building, 240 Pasteur ***@***.***,-122.1795782,17z/data=!4m5!3m4!1s0x808fa52ba51d58eb:0xe8daaea65a01418c!8m2!3d37.4329555!4d-122.1773895>
Stanford, CA 94305   ***@***.***,-122.1795782,17z/data=!4m5!3m4!1s0x808fa52ba51d58eb:0xe8daaea65a01418c!8m2!3d37.4329555!4d-122.1773895>
Personal website: shenxt.info<https://www.shenxt.info/>
Phone: +1 571 267 9283

From: Laurent Gatto ***@***.***>
Date: Tuesday, January 25, 2022 at 05:24
To: Bioconductor/Contributions ***@***.***>
Cc: Xiaotao Shen ***@***.***>, Mention ***@***.***>
Subject: Re: [Bioconductor/Contributions] massconverter (Issue #2482)

Hi @jaspershen<https://github.com/jaspershen> - I am reviewing your packages and wanted to make sure I don't miss any. I'm aware of massconverter and masstools (issue #2481<https://github.com/Bioconductor/Contributions/issues/2481>); anything else?

**Comment by vjcitn** [comment]
The vignettes of massconverter are seemingly designed for pkgdown.  I find it difficult to navigate them through help.start().  It is specifically unclear whether the user HAS TO use docker with this package.  Please write an overview.Rmd that explains concisely what the package does and what the user must do to use the package.  Note that the vast majority of packages in Bioconductor do not require docker client usage.  If it is in fact mandatory for your package we need to discuss this further.

**Comment by vjcitn** [comment]
This is all nice but not relevant for a Bioconductor submission:

![massc](https://user-images.githubusercontent.com/4624066/155351296-a5ca9075-e6ab-4d55-837c-9dabe0c1eddd.jpg)

Review cannot begin until the vignettes are more consistent with the Bioconductor guidelines.  Thanks.

**Comment by vjcitn** [comment]
Please add prefixes to your vignette titles so they are ordered meaningfully in the page generated by help.start()

**Comment by lgatto** [comment]
Hi @jaspershen - here are a few comments to consider before going to a detailed review:
1. The vignettes need to be amended to fit a Bioconductor package rather than a pkgdown webpage. This is also true for [masstools](https://github.com/Bioconductor/Contributions/issues/2481).
2. There needs to be an example that is actually run. Currently, the code in the vignette that is supposed to convert raw data isn't evaluated (even though there's a RAW file in the data), and actually fails. It isn't clear if the data files need to be in a specific directory (looks like based on the error message), but it wasn't obvious to me and I couldn't find the relevant info in the manual page. 
3. Finally, and most importantly, I am not convinced that `massconverter` is a good fit for Bioconductor. That doesn't mean that the package isn't useful, of course; it is an interface to pull and run docker container to convert raw MS data directly from R, but it doesn't really address any goal of the Bioconductor project (analysis and comprehension of biological data). Tagging @lshep and @vjcitn to see what they think.

**Comment by vjcitn** [comment]
Sorry to have missed this.  I would tend to agree that software that *requires* docker is out of scope for Bioconductor at present.  I will look more closely at massconverter and comment further.

**Comment by vjcitn** [comment]
Is massconverter per se essential to other packages in the tidymass ecosystem?  I think its reliance on docker client separates it from all other bioconductor packages to date.  I don't want to be technologically restrictive but I think reliance on something like `https://hub.docker.com/r/chambm/pwiz-skyline-i-agree-to-the-vendor-licenses` is very unappealing.

**Comment by vjcitn** [comment]
@jaspershen will you be responding on these review issues?  @lshep

**Comment by jaspershen** [comment]
Hi@vjcitn. Thank you for the noticement. I will respond very this week. Sorry for the late reply! Thank you!
